jbmc/unit/java_bytecode/java_static_initializers/assignments_from_json.cpp

```diff
@@ -0,0 +1,412 @@
+      field_symbol.type = java_int_type();
+      field_symbol.is_static_lifetime = true;
+      return field_symbol;
+    }());
+    const irep_idt clinit_name = add_clinit(symbol_table, "TestClass");
+    add_class_with_components(
+      symbol_table,
+      "TestClass",
+      {std::pair<std::string, typet>("@class_identifier", string_typet{}),
+       std::pair<std::string, typet>("array_field", java_array_type('i'))});
+    // For array[int]
+    add_java_array_types(symbol_table);
+
+    const reference_typet test_class_type =
+      reference_type(struct_tag_typet{"java::TestClass"});
+    code_blockt block;
+    optionalt<ci_lazy_methods_neededt> option{};
+    assign_from_json(
+      symbol_exprt{"symbol_to_assign", test_class_type},
+      json,
+      clinit_name,
+      block,
+      symbol_table,
+      option,
+      max_user_array_length,
+      references);
+
+    THEN("The instruction is the declaration of a symbol of TestClass* type")
+    {
+      const symbol_exprt allocation_symbol =
+        make_query(block)[0].as<code_declt>()[0].as<symbol_exprt>().get();
+      REQUIRE(allocation_symbol.type() == test_class_type);
+      THEN("The instruction declares a symbol for the length")
+      {
+        REQUIRE(
+          make_query(block)[1]
+            .as<code_declt>()[0]
+            .as<symbol_exprt>()
+            .get()
+            .type() == java_int_type());
+      }
+      THEN("The instruction declares the array data")
+      {
+        REQUIRE(
+          make_query(block)[2]
+            .as<code_declt>()[0]
+            .as<symbol_exprt>()
+            .get()
+            .type() == java_reference_type(java_int_type()));
+      }
+      THEN("The instruction allocates the struct")
+      {
+        REQUIRE(
+          make_query(block)[3].as<code_assignt>()[0].as<symbol_exprt>().get() ==
+          allocation_symbol);
+      }
+    }
+    THEN("The instruction assigns the symbol to \"symbol_to_assign\"")
+    {
+      REQUIRE(
+        make_query(block)[4]
+          .as<code_assignt>()[0]
+          .as<symbol_exprt>()
+          .get()
+          .get_identifier() == "symbol_to_assign");
+    }
+
+    THEN("The instruction zero-initializes the struct")
+    {
+      REQUIRE(
+        make_query(block)[5].as<code_assignt>()[0].get().type() ==
+        test_class_type.subtype());
+    }
+    THEN("The instruction makes the length nondet")
+    {
+      REQUIRE(
+        make_query(block)[6].as<code_assignt>()[0].get().type() ==
+        java_int_type());
+      auto side_effect = make_query(block)[6]
+                           .as<code_assignt>()[1]
+                           .as<side_effect_exprt>()
+                           .get();
+      REQUIRE(side_effect.get_statement() == ID_nondet);
+    }
+    THEN("The instruction adds assumption on the length")
+    {
+      REQUIRE(
+        make_query(block)[7].as<code_assumet>()[0].get().type() ==
+        bool_typet{});
```

Comment:
:bulb: Can we check what the assumptions for this are, rather than just check that they have boolean type?

Reply:
that's difficult to do in a robust manner because you can have different assumptions which are logically equivalent...